src/main/AndroidManifest.xml

```diff
@@ -51,6 +51,20 @@
                 android:value=".activities.MainActivity" />
         </activity>
         <activity
+            android:name=".activities.ShareActivity"
+            android:label="@string/share_activity_title">
+            <intent-filter>
+                <action android:name="android.intent.action.SEND" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:mimeType="*/*" />
+            </intent-filter>
+            <intent-filter>
+                <action android:name="android.intent.action.SEND_MULTIPLE" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:mimeType="*/*" />
```

Comment:
When you share plain text file - you get its contents from `Intent.EXTRA_STREAM` which will be a url. You can then get file name based on schema.
When you share some text or memo the contents will be inside `Intent.EXTRA_TEXT`.
And both these cases will be with `mimeType = 'text/plain'`. You can't pre-filter intent by EXTRA_* fields.
Google Drive handles everything. And that's what I want too.